src/StoreApi/Routes/Checkout.php

```diff
@@ -170,8 +172,9 @@ protected function get_route_post_response( \WP_REST_Request $request ) {
 
 		// Create a new user account as necessary.
 		try {
-			// TODO full docs for new hook, including throw.
-			do_action( 'woocommerceblocks_create_account_for_order_request', $order_object, $request );
+			$create_account = Package::container()->get( CreateAccount::class );
+			$create_account->from_order_request( $order_object, $request );
+
 		} catch ( Exception $error ) {
 			$this->handle_create_account_error( $error );
```

Comment:
Rather than being _specifically_ for handling create account errors, I wonder if this should just be `handle_error`? That way it's not tightly coupled to errors related to create account (and feasibly other error codes could be handled by the method in the future).

Reply:
I can work with that, made the change here: 65a72403f7dda3b9eb84494f3abf545a179288db

Incidentally, I went with the specific name as it's private – can easily rename later when generalising - and to more clearly indicate the intention (since it's not intended to handle any error right now).